starter-core/src/main/java/io/micronaut/starter/build/dependencies/MicronautDependencyUtils.java

```diff
@@ -268,21 +268,45 @@ public static Dependency.Builder annotationProcessor(@NonNull BuildTool buildToo
                     .groupId(groupId)
                     .artifactId(artifactId)
                     .annotationProcessor();
-            case MAVEN -> moduleMavenAnnotationProcessor(groupId, artifactId, propertyName, requiresPriority);
+            case MAVEN -> moduleMavenAnnotationProcessor(groupId, artifactId, propertyName, false, requiresPriority);
+        };
+    }
+
+    @NonNull
+    public static Dependency.Builder testAnnotationProcessor(@NonNull BuildTool buildTool,
+                                                             @NonNull String groupId,
+                                                             @NonNull String artifactId,
+                                                             @NonNull String propertyName) {
+        return testAnnotationProcessor(buildTool, groupId, artifactId, propertyName, false);
+    }
+
+    @NonNull
+    public static Dependency.Builder testAnnotationProcessor(@NonNull BuildTool buildTool,
+                                                         @NonNull String groupId,
+                                                         @NonNull String artifactId,
+                                                         @NonNull String propertyName,
+                                                         boolean requiresPriority) {
+        return switch (buildTool) {
+            case GRADLE, GRADLE_KOTLIN -> Dependency.builder()
+                    .groupId(groupId)
+                    .artifactId(artifactId)
+                    .testAnnotationProcessor();
+            case MAVEN -> moduleMavenAnnotationProcessor(groupId, artifactId, propertyName, true, requiresPriority);
         };
     }
 
     @NonNull
     public static Dependency.Builder moduleMavenAnnotationProcessor(@NonNull String groupId,
                                                                     @NonNull String artifactId,
                                                                     @NonNull String propertyName,
+                                                                    boolean isTestScope,
                                                                     boolean requiresPriority) {
-        return Dependency.builder()
+        Dependency.Builder dependency = Dependency.builder()
                 .groupId(groupId)
                 .artifactId(artifactId)
-                .annotationProcessor(requiresPriority)
-                .versionProperty(propertyName)
-                .exclude(MICRONAUT_INJECT);
```

Comment:
why did we remove `.exclude(MICRONAUT_INJECT); here ? @alvarosanchez @wetted ?

Reply:
No longer necessary, for any kind of project. I still need to review other places where we still exclude it

Reply:
> No longer necessary, for any kind of project. I still need to review other places where we still exclude it

And for groovy/maven projects we were getting duplicate dependencies anyway, one with the excludes and one without.